rllib/train.py

```diff
@@ -257,14 +261,30 @@ def patch_path(path):
             )
         ray.init(address=cluster.address)
     else:
-        ray.init(
-            include_dashboard=args.ray_ui,
-            address=args.ray_address,
-            object_store_memory=args.ray_object_store_memory,
-            num_cpus=args.ray_num_cpus,
-            num_gpus=args.ray_num_gpus,
-            local_mode=args.local_mode,
-        )
+        # Try connecting to given address (by default, this is "auto").
+        try:
+            ray.init(
+                include_dashboard=not args.no_ray_ui,
+                address=args.ray_address,
+                object_store_memory=args.ray_object_store_memory,
+                num_cpus=args.ray_num_cpus,
+                num_gpus=args.ray_num_gpus,
+                local_mode=args.local_mode,
+            )
+        # Last resort: Try running in a new ray cluster (address=None).
+        except ConnectionError:
+            logger.warning(
+                "Connecting to ray-address={} not successful! Trying to run in a "
+                "new ray cluster."
+            )
```

Comment:
Hmm, this will be the common case. I don't think we want to make all laptop users see this warning.